jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Jps.java

```diff
@@ -68,38 +83,52 @@ public static void main(String[] args) {
 		System.exit(rc);
 	}
 	
-	@SuppressWarnings("nls")
 	private static void parseArguments(String[] args) {
 		printApplicationArguments = false;
 		printJvmArguments = false;
 		noPackageName = true;
 		vmidOnly = false;
-		final String HELPTEXT = "jps: Print a list of Java processes and information about them%n"
-				+ "    -J: supply arguments to the Java VM running jps%n"
-				+ "    -l: print the application package name%n"
-				+ "    -q: print only the virtual machine identifiers%n"
-				+ "    -m: print the application arguments%n"
-				+ "    -v: print the Java VM arguments, including those produced automatically%n";
-		for (String a : args) {
-			switch (a) {
-			case "-l":
+
+		for (String arg : args) {
+			if (!arg.startsWith("-") || arg.length() == 1) {
+				System.out.printf("malformed argument '%s'%n%n", arg);
```

Comment:
Since this PR intends to match `RI` behaviours, this error message can be modified to `illegal argument:` as well such that the script doesn't have to handle different errors.

@pshipton any thoughts?

Reply:
Makes sense

Reply:
Improved